saleor/graphql/checkout/mutations/checkout_complete.py

```diff
@@ -164,6 +167,43 @@ def validate_checkout_addresses(
         if billing_address_data != billing_address.as_data():
             billing_address.save()
 
+    @classmethod
+    def _reserve_stocks_for_checkout_complete(
+        cls,
+        checkout_info: CheckoutInfo,
+        lines: Iterable[CheckoutLineInfo],
+        site_settings,
+    ):
+        """Add temporary reservation for stocks, to get rid of locking stock rows."""
+        RESERVE_DURATION = 45
+
+        variants = [line.variant for line in lines]
+        stocks = Stock.objects.get_variants_stocks_for_country(
+            country_code=checkout_info.get_country(),
+            channel_slug=checkout_info.channel,
+            products_variants=variants,
+        )
+        variants_stocks_map = {stock.product_variant_id: stock for stock in stocks}
+
+        dummy_reservations = []
+        for line in lines:
+            if line.variant.id in variants_stocks_map:
+                dummy_reservations.append(
+                    Reservation(
+                        quantity_reserved=line.line.quantity,
+                        reserved_until=timezone.now()
+                        + timedelta(seconds=RESERVE_DURATION),
+                        stock=variants_stocks_map[line.variant.id],
+                        checkout_line=line.line,
+                    )
+                )
+        Reservation.objects.bulk_create(dummy_reservations)
```

Comment:
I don't like that we now have multiple implementations of reservation creation. There should be a common set of helpers like the ones for increasing/decreasing stock shared between this and reservations created for checkout.

Reply:
The thing is, existing `def reserve_stocks()` function, include stock availability check. But in some cases, with the additional reservation approach, we we will exceed number of stocks available and `reserve_stock` function will return an `InsufficientStock` error.

Reply:
We want to validate stock in the first transaction. So the atomic operation is two-step: validate stock (locking it, so we know it's valid) and create reservation (while the stock is locked to avoid two processes reserving the same stock), then unlock both.

Reply:
Validation is done in the first transaction, here: `_get_order_data` -> `_prepare_order_data` -> `_create_lines_for_order`-> `check_stock_and_preorder_quantity_bulk(..., check_reservations=True)`